src/Library/demos/Drop Zone/main.js

```diff
@@ -0,0 +1,124 @@
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio";
+import Adw from "gi://Adw";
+import Gdk from "gi://Gdk";
+import GObject from "gi://GObject";
+
+const bin = workbench.builder.get_object("bin");
+
+// Universal drop target for any String data
+const string_drop_target = Gtk.DropTarget.new(
+  GObject.TYPE_STRING,
+  Gdk.DragAction.COPY,
+);
+
+bin.add_controller(string_drop_target);
+
+string_drop_target.connect("drop", (self, value, x, y) => {
+  bin.child = createTextWidget(value);
+  bin.get_style_context().remove_class("overlay-drag-area");
+});
+
+// Drop Target for Files
+const file_drop_target = Gtk.DropTarget.new(Gio.File, Gdk.DragAction.COPY);
+bin.add_controller(file_drop_target);
+
+file_drop_target.connect("drop", (self, value, x, y) => {
+  if (!(value instanceof Gio.File)) return false;
+
+  const file_info = value.query_info("standard::content-type", 0, null);
+  const content_type = file_info.get_content_type();
+
+  if (content_type.startsWith("image/")) {
+    try {
+      bin.child = createImageWidget(value);
+    } catch (error) {
+      console.log(`Unable to load image: ${error}`);
```

Comment:
always use `logError` in GJS (at least for now, until we fix `console` to deal with it properly)